build_tools/docker/cmake-bazel-frontends/Dockerfile

```diff
@@ -24,3 +24,10 @@ RUN python3 -m pip install tf-nightly==2.6.0.dev20210412
 
 # Install JAX.
 RUN python3 -m pip install --upgrade jax jaxlib flax
+
+# Install KWS Streaming Library.
+RUN python3 -m pip install --upgrade tensorflow-model-optimization==0.5.1.dev0
+ARG KWS_COMMIT=4bfbfe4ec19fe39cf9d1803e894d8bfb5fd56a45
+RUN git clone https://github.com/google-research/google-research
+RUN cd google-research && git checkout "${KWS_COMMIT?}" && cd ..
+ENV PYTHONPATH="/google-research:$PYTHONPATH"
```

Comment:
Can we avoid adding all of google-research to the image and to the PYTHONPATH please? That unnecessarily bloats the image and makes it possible for tests to randomly pull in stuff from google-research, which I don't think we want (probably also slows them down while they search, but I'd bet that's not a big factor).

Instead we can fetch a tar archive. I of course can't leave a suggested edit on unchanged lines because GitHub is super cool, but here's what I'd recommend.

As a previous stage:

```dockerfile
FROM gcr.io/iree-oss/util@sha256:deadbeef AS fetch-kws
WORKDIR /fetch-kws

ARG KWS_COMMIT=4bfbfe4ec19fe39cf9d1803e894d8bfb5fd56a45

wget "https://github.com/google-research/google-research/tarball/${KWS_COMMIT?}" -O google-research.tar.gz
tar --extract --gzip --file=google-research.tar.gz --wildcards */kws_streaming --strip-components=1
```

And then in the current stage


```suggestion
# Install KWS Streaming Library.
RUN python3 -m pip install --upgrade tensorflow-model-optimization==0.5.1.dev0

COPY --from=fetch-kws /fetch-kws/kws_streaming /local-python/kws_streaming
ENV PYTHONPATH="/local-python:$PYTHONPATH"
```

Where you can call local-python whatever you want if there's some other convention.

Reply:
There aren't any conventions that I know of, but I doubt I would know if there were.